src/sentry/tasks/merge.py

```diff
@@ -257,11 +257,16 @@ def merge_objects(models, group, new_group, limit=1000,
                             )
                     elif model == GroupTagValue:
                         with transaction.atomic(using=router.db_for_write(model)):
```

Comment:
Because the `update` may fail, and we need to then roll back this instead of rolling back the larger transaction. `times_seen` might overflow.